src/test/java/org/generationcp/middleware/service/impl/inventory/LotServiceImplIntegrationTest.java

```diff
@@ -0,0 +1,175 @@
+
+		final TransactionsSearchDto discardedTransactionSearch = new TransactionsSearchDto();
+		discardedTransactionSearch.setLotIds(Collections.singletonList(lot.getId()));
+		discardedTransactionSearch.setTransactionTypes(Collections.singletonList(TransactionType.DISCARD.getId()));
+		final List<TransactionDto> discardedTransactions = transactionService.searchTransactions(discardedTransactionSearch, null);
+		discardedTransactions.sort((TransactionDto t1, TransactionDto t2) -> t2.getTransactionId().compareTo(t1.getTransactionId()));
+
+		assertThat(extendedLotDtos.get(0).getStatus(), hasToString(LotStatus.CLOSED.name()));
+		assertThat(extendedLotDtos.get(0).getAvailableBalance(), equalTo(0D));
+		assertThat(pendingTransactions, hasSize(0));
+		assertThat(discardedTransactions.get(0).getAmount(), equalTo(-20D));
+	}
+
+	@Test
+	public void lotWithNoOpenBalanceClosed_Ok() {
+		final Transaction confirmedWithdrawal =
+			new Transaction(null, userId, lot, Util.getCurrentDate(), TransactionStatus.CONFIRMED.getIntValue(),
+				-20D, "Transaction 3", 0, null, null, null,
+				Double.valueOf(0), userId, TransactionType.WITHDRAWAL.getId());
+		this.daoFactory.getTransactionDAO().save(confirmedWithdrawal);
+
+		final TransactionsSearchDto discardedTransactionSearch = new TransactionsSearchDto();
+		discardedTransactionSearch.setLotIds(Collections.singletonList(lot.getId()));
+		discardedTransactionSearch.setTransactionTypes(Collections.singletonList(TransactionType.DISCARD.getId()));
+		final Integer discardedTrxsBeforeClosingLot = this.transactionService.searchTransactions(discardedTransactionSearch, null).size();
+
+		this.lotService.closeLots(userId, Collections.singletonList(lot.getId()));
+		final Integer discardedTrxsAfterClosingLot = this.transactionService.searchTransactions(discardedTransactionSearch, null).size();
+
+		final LotsSearchDto searchDto = new LotsSearchDto();
+		searchDto.setLotIds(Collections.singletonList(lot.getId()));
+		final List<ExtendedLotDto> extendedLotDtos = this.lotService.searchLots(searchDto, null);
+
+		final TransactionsSearchDto pendingTransactionSearch = new TransactionsSearchDto();
+		pendingTransactionSearch.setLotIds(Collections.singletonList(lot.getId()));
+		pendingTransactionSearch.setTransactionStatus(Collections.singletonList(TransactionStatus.PENDING.getIntValue()));
+		final List<TransactionDto> pendingTransactions = transactionService.searchTransactions(pendingTransactionSearch, null);
+
+		assertThat(extendedLotDtos.get(0).getStatus(), hasToString(LotStatus.CLOSED.name()));
+		assertThat(extendedLotDtos.get(0).getAvailableBalance(), equalTo(0D));
+		assertThat(pendingTransactions, hasSize(0));
+		assertThat(discardedTrxsAfterClosingLot, equalTo(discardedTrxsBeforeClosingLot));
+
+	}
+
+	private void findAdminUser() {
+		final WorkbenchUser user = this.userService.getUserByName("admin", 0, 1, Operation.EQUAL).get(0);
+		userId = user.getUserid();
+	}
```

Comment:
This might be part of IntegrationTestBase

Reply:
done